tests/drivers/ipm/src/main.c

```diff
@@ -34,7 +34,7 @@ extern struct ipm_driver_api ipm_dummy_api;
 struct ipm_dummy_driver_data ipm_dummy0_driver_data;
 DEVICE_DEFINE(ipm_dummy0, "ipm_dummy0", NULL,
 		NULL, &ipm_dummy0_driver_data, NULL,
-		POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+		PRE_KERNEL_2, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
```

Comment:
why PRE_KERNEL_2? I would avoid this and use PRE_KERNEL_1 if possible, PRE_KERNEL_2 is something that hopefully will go away soon.

Reply:
> why PRE_KERNEL_2?

Looking at drivers/ipm where both are used, my choice was the closest to `POST_KERNEL` as it was here.
There is no difference when this test runs on `qemu`-s.

> PRE_KERNEL_2 is something that hopefully will go away soon.

interesting,  I didn't come across this info in the sources or docs, eg.
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/kernel/drivers/index.rst#initialization-levels

Reply:
it is being discussed, please move this to PRE_KERNEL_1 for now.

Reply:
ok, changed to PRE_KERNEL_1